src/evse_monitor.cpp

```diff
@@ -266,6 +267,8 @@ void EvseMonitor::evseBoot(const char *firmware)
     }
 
   });
+
+  incrementConfig();
```

Comment:
The config should listen in to the evse.onBootReady event so there is not any circular dependancy

Reply:
I'm a little lost with this listener logic: where can I place the call exactly ?

Reply:
Have A look at [input.cpp](https://github.com/OpenEVSE/ESP32_WiFi_V4.x/blob/master/src/input.cpp#L27)

Reply:
Also FYI all the above calls are asyncronous so may not have cmopleted at this point. The boot event won't suffer from that at it is only triggered after all calls have been completed

Reply:
Ok I understand! I'll need to create a micro task in the app config!

Reply:
@KipK @jeremypoulter : I re-thought it again and we actually don't need to listen on anything.

I am initialising the config version to 1 internally, and 0 in the mqtt.

in mqtt, I am cheking: `evse.getEvseState() == OPENEVSE_STATE_STARTING`. The method `mqtt_publish_config` returns true if it wa sable to publish.

Then, in the mqtt loop:

```c
    if(configVersion != config_version() && mqtt_publish_config()) {
      DBUGF("Config has changed, publishing to MQTT");
      configVersion = config_version();
    }
```

So after a firmware flash or complete power loss / back up again, the loop will keep trying to republish the config until mqtt is connected and evse is started.